### DISCUSSION TITLE
Custom style URL not showing custom styles consistently after offline download

### DISCUSSION THREAD

**Discussion by kevinmanncito** [rnmapbox/maps#2856] [Q&A]
Hey @mfazekas, this V10 version is working great for us! Thanks for all the hard work. However, there is the one issue we have run into.. Here it is:

* Download an offline map with custom styles
* After downloading the custom styles disappear at various zoom intervals (seems to mostly affect linestrings)
* Before download there is no issue with the custom style at any zoom level

I have been able to reproduce this with a super simple custom style: mapbox://styles/onwaterllc/clhcm7yjv00dl01rh4d0thgaz

Here is my fork of the rnmapbox/maps examples app that also reproduces the issue: https://github.com/rnmapbox/maps/compare/main...kevinmanncito:maps:offline-map-demo

Here is a demo video of me zooming around before downloading and the custom styles seem to work fine:

https://github.com/rnmapbox/maps/assets/2521298/81524075-f3a8-405e-93a2-a1a467e6d08f

Here is a demo video of me zooming around after downloading and the issue appears:

https://github.com/rnmapbox/maps/assets/2521298/cd5510af-7abd-4bc4-a628-770160728d24

I'm happy to fill out the issue template as well if that would be best, just let me know! Thanks

**Comment by kevinmanncito** [comment]
Using the example app for the native ios sdk I was able to reproduce this issue upstream. I'll post over there and see what I can find.

https://github.com/rnmapbox/maps/assets/2521298/f3519fa4-6dc6-4463-bcb4-014df017a92b

**Comment by kevinmanncito** [comment]
And... here is the upstream issue that was already there: https://github.com/mapbox/mapbox-maps-ios/issues/1940

**Comment by mfazekas** [comment]
@kevinmanncito awesome, thanks much for all the work on this!

**Comment by kevinmanncito** [comment]
I found a decent workaround for our use case (thanks to mapbox support). Turn off source compositing in the custom style:

![Screenshot 2023-05-16 at 10 23 30 AM](https://github.com/mapbox/mapbox-maps-ios/assets/2521298/e1a808f5-071e-4a1d-a75d-51e75b44e00f)

Make sure to do that first thing when you create the map then add the custom sources to the map.

It sounds like there is a slight performance degradation when source compositing is off but it hasn't been noticeable for us. Here is the link to the docs on source compositing: https://docs.mapbox.com/studio-manual/reference/styles/#source-compositing
